services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy

```diff
@@ -410,77 +412,162 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
     @Tidy
     @Tags(VIRTUAL)
     def "Flow in 'Down' status is rerouted after switchUp event"() {
-        given: "Two active neighboring switches with two parallel links and two available paths"
+        given: "First switch pair with two parallel links and two available paths"
         assumeTrue("Reroute should be completed before link is FAILED", rerouteDelay * 2 < discoveryTimeout)
-        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+        def switchPair1 = topologyHelper.getAllNeighboringSwitchPairs().find {
             it.paths.findAll { it.size() == 2 }.size() > 1
-        } ?: assumeTrue("No suiting switches found", false)
+        } ?: assumeTrue("No suiting switches found for the first flow", false)
+
+        and: "Second switch pair where the scr switch from the first switch pair is a transit switch"
+        List<PathNode> secondFlowPath
+        def switchPair2 = topologyHelper.getAllNotNeighboringSwitchPairs().find { swP ->
+            swP.paths.find { pathCandidate ->
+                secondFlowPath = pathCandidate
+                def involvedSwitches = pathHelper.getInvolvedSwitches(pathCandidate)
+                involvedSwitches.size() == 3 && involvedSwitches[1].dpId == switchPair1.src.dpId &&
+                        involvedSwitches[-1].dpId == switchPair1.dst.dpId
```

Comment:
Because of this condition I'm afraid you'll have to include all reversed(mirrored) switch pairs during search. Because all remaining switch pairs may use `switchPair1.dst.dpId` as their src

Reply:
thx, fixed